### ISSUE TITLE
String index out of range

### ISSUE THREAD

**Issue by lasekio** [johnlindquist/AceJump#136]
Sometimes ace jump just flicks once and disappear after 1s so i cannot use it. This is error, that shows in the console: 

```
String index out of range: 115
java.lang.StringIndexOutOfBoundsException: String index out of range: 115
	at java.lang.String.charAt(String.java:658)
	at com.johnlindquist.acejump.search.Finder.maybeJump(Finder.kt:74)
	at com.johnlindquist.acejump.search.Finder.findOrJump(Finder.kt:44)
	at com.johnlindquist.acejump.KeyboardHandler$findAndUpdateUI$1.run(KeyboardHandler.kt:49)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:926)
	at com.johnlindquist.acejump.KeyboardHandler.findAndUpdateUI(KeyboardHandler.kt:49)
	at com.johnlindquist.acejump.KeyboardHandler.access$findAndUpdateUI(KeyboardHandler.kt:31)
	at com.johnlindquist.acejump.KeyboardHandler$interceptPrintableKeystrokes$1.execute(KeyboardHandler.kt:101)
	at com.intellij.openapi.editor.actionSystem.TypedAction.lambda$actionPerformed$0(TypedAction.java:142)
	at com.intellij.reporting.FreezeLoggerImpl.runUnderPerformanceMonitor(FreezeLoggerImpl.java:42)
	at com.intellij.openapi.editor.actionSystem.TypedAction.actionPerformed(TypedAction.java:142)
	at com.intellij.openapi.editor.impl.EditorImpl.processKeyTypedNormally(EditorImpl.java:1114)
	at com.intellij.openapi.editor.impl.EditorImpl.a(EditorImpl.java:1102)
	at com.intellij.openapi.editor.impl.EditorImpl.processKeyTyped(EditorImpl.java:3115)
	at com.intellij.openapi.editor.impl.EditorImpl$6.keyTyped(EditorImpl.java:937)
	at java.awt.AWTEventMulticaster.keyTyped(AWTEventMulticaster.java:239)
	at java.awt.Component.processKeyEvent(Component.java:6496)
	at javax.swing.JComponent.processKeyEvent(JComponent.java:2832)
	at java.awt.Component.processEvent(Component.java:6318)
	at java.awt.Container.processEvent(Container.java:2236)
	at java.awt.Component.dispatchEventImpl(Component.java:4897)
	at java.awt.Container.dispatchEventImpl(Container.java:2294)
	at java.awt.Component.dispatchEvent(Component.java:4719)
	at java.awt.KeyboardFocusManager.redispatchEvent(KeyboardFocusManager.java:1954)
	at java.awt.DefaultKeyboardFocusManager.dispatchKeyEvent(DefaultKeyboardFocusManager.java:806)
	at java.awt.DefaultKeyboardFocusManager.preDispatchKeyEvent(DefaultKeyboardFocusManager.java:1074)
	at java.awt.DefaultKeyboardFocusManager.typeAheadAssertions(DefaultKeyboardFocusManager.java:945)
	at java.awt.DefaultKeyboardFocusManager.dispatchEvent(DefaultKeyboardFocusManager.java:771)
	at com.intellij.ide.IdeKeyboardFocusManager.dispatchEvent(IdeKeyboardFocusManager.java:27)
	at java.awt.Component.dispatchEventImpl(Component.java:4768)
	at java.awt.Container.dispatchEventImpl(Container.java:2294)
	at java.awt.Window.dispatchEventImpl(Window.java:2746)
	at java.awt.Component.dispatchEvent(Component.java:4719)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:764)
	at java.awt.EventQueue.access$500(EventQueue.java:98)
	at java.awt.EventQueue$3.run(EventQueue.java:715)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:80)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:90)
	at java.awt.EventQueue$4.run(EventQueue.java:737)
	at java.awt.EventQueue$4.run(EventQueue.java:735)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:80)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:734)
	at com.intellij.ide.IdeEventQueue.d(IdeEventQueue.java:827)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:634)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:365)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
```

**Comment by lasekio** [comment]
![zrzut ekranu 2017-05-19 o 13 18 57](https://cloud.githubusercontent.com/assets/8137000/26245716/ba9a1670-3c95-11e7-86c1-63a80f51a3e7.png)

**Comment by breandan** [comment]
I suspect this is caused by some weird edge case buried deep in [this mess](https://github.com/johnlindquist/AceJump/blob/master/src/main/kotlin/com/johnlindquist/acejump/search/Finder.kt#L61). Investigating, thanks for reporting.

**Comment by breandan** [comment]
Fixed in 726f3d2, thank you for reporting! @amandle @lasekio #150
